### PR TITLE
feat (img) emit ionImgDidLoad only when image is actually loaded

### PR DESCRIPTION
**Pull request by vhinic**
emit ionImgIsLoading when img src is set and emit ionImgDidLoad only when image is loaded

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

In the current implementation ionImgDidLoad is emitted as soon as src is set. 

Issue Number: #18161

## What is the new behavior?

- `ionImgIsLoading` is emitted when img src is set
- `ionImgDidLoad` is emitted only after actual img is loaded

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

`ionImgDidLoad ` is not emitted when src is set but rather when img is actually loaded

### CODE REVIEW

File: core/src/components/img/img.tsx [hunk 1]
```diff
@@ -35,6 +35,9 @@ export class Img implements ComponentInterface {
   }
 
   /** Emitted when the img src has been set */
+  @Event() ionImgIsLoading!: EventEmitter<void>;
```
Comment (liamdebeasi) [inline core/src/components/img/img.tsx]: Let's change `ionImgIsLoading` to `ionImgWillLoad`

File: core/src/components/img/img.tsx [hunk 2]
```diff
@@ -35,6 +35,9 @@ export class Img implements ComponentInterface {
...
+
+  /** Emitted when the img has been loaded */
```
Comment (liamdebeasi) [inline core/src/components/img/img.tsx]: Let's have this say "Emitted when the image has finished loading."

**Comment by brandyscarney** [comment]
I would not consider it a breaking change to fix this.

**Comment by brandyscarney** [comment]
Thank you for the PR!

**Pull request merged**